tests/integration/cache_test.go

```diff
@@ -487,22 +487,77 @@ func TestCacheWithPrefixWatch(t *testing.T) {
 }
 
 func TestCacheWithoutPrefixGet(t *testing.T) {
-	integration.BeforeTest(t)
-	clus := integration.NewCluster(t, &integration.ClusterConfig{Size: 1})
-	t.Cleanup(func() { clus.Terminate(t) })
-
-	client := clus.Client(0)
+	lastRev := TestGetEvents[len(TestGetEvents)-1].Kv.ModRevision
 
-	c, err := cache.New(client, "", cache.WithHistoryWindowSize(32))
-	if err != nil {
-		t.Fatalf("cache.New: %v", err)
-	}
-	t.Cleanup(c.Close)
-	if err := c.WaitReady(t.Context()); err != nil {
-		t.Fatalf("cache.WaitReady: %v", err)
+	tcs := []struct {
+		name                string
+		initialLoadRevision int64
+	}{
+		{"empty cache", 0},
+		{"partial load", 4},
+		{"complete load", lastRev},
 	}
 
-	testGet(t, client.KV, c)
+	for _, tc := range tcs {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			integration.BeforeTest(t)
+			clus := integration.NewCluster(t, &integration.ClusterConfig{Size: 1})
+			t.Cleanup(func() { clus.Terminate(t) })
+			client := clus.Client(0)
+			kv := client.KV
+			ctx := t.Context()
+
+			t.Log("Classify and load first batch of events")
+			var firstBatch, secondBatch []*clientv3.Event
+			for _, ev := range TestGetEvents {
+				if ev.Kv.ModRevision <= tc.initialLoadRevision {
+					firstBatch = append(firstBatch, ev)
+				} else {
+					secondBatch = append(secondBatch, ev)
+				}
+			}
```

Comment:
Maybe instead of doing filtering here do it in test cases

tcs := []struct {
		name                string
		initialEvents, followupEvents []*clientv3.Event
	}{
		{"empty cache", nil, TestGetEvents},
		{"partial load", filterEvents(TestGetEvents, revisionLessThan(4)), filterEvents(TestGetEvents, revisionGreatherEqual(4))},
		{"complete load", TestGetEvents, nil},
	}